### PR TITLE
Rename all instances of stop tactic to halt tactic

### LINKED ISSUE #2630: Rename StopTactic to HaltTactic
### Description of the task

StopTactic is too easily confused with the StopPlay, which doesn't involve halting all the robots. Renaming StopTactic to HaltTactic would be more consistent with the HaltPlay.

### Acceptance criteria

- [ ] Rename StopTactic to HaltTactic

### PR DESCRIPTION
**Pull request by someone2060**
### Description

Renames all instances of stop tactic to halt tactic. ~~`tactic.proto` is now inconsistent with `HaltTactic` with this refactoring.~~ `tactic.proto` inconsistency has been fixed.

#3037 is a PR of the same issue, but was not merged.

### Testing Done

`halt_fsm_test`, `halt_tactic_test`, and `play_selection_fsm_test` tests were run.

### Resolved Issues

resolves #2630 

### Length Justification and Key Files to Review

Files in [this folder](https://github.com/UBC-Thunderbots/Software/tree/3d7810f6e126c34f499bbebf8f6b33d2cddcec55/src/software/ai/hl/stp/tactic/halt) have been renamed.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/proto/tactic.proto [hunk 1]
```diff
@@ -28,7 +28,7 @@ message Tactic
         PivotKickTactic pivot_kick                          = 11;
         ReceiverTactic receiver                             = 12;
         ShadowEnemyTactic shadow_enemy                      = 13;
-        StopTactic stop                                     = 14;
+        HaltTactic stop                                     = 14;
```
Comment (Andrewyx) [inline src/proto/tactic.proto]: This should be ideally changed to `halt`

File: src/software/field_tests/movement_robot_field_test.py [hunk 2]
```diff
@@ -117,10 +117,10 @@ def test_basic_rotation(field_test_runner):
...
             test_timeout_s=5,
         )
-        # Send a stop tactic after the test finishes
-        stop_tactic = StopTactic()
+        # Send a halt tactic after the test finishes
+        halt_tactic = HaltTactic()
         params = AssignedTacticPlayControlParams()
-        params.assigned_tactics[id].stop.CopyFrom(stop_tactic)
+        params.assigned_tactics[id].stop.CopyFrom(halt_tactic)
         # send the stop tactic
```
Comment (Andrewyx) [inline src/software/field_tests/movement_robot_field_test.py]: Update comments

File: src/proto/tactic.proto (hunk 1 above)
Comment (someone2060) [reply src/proto/tactic.proto]: fixed

File: src/software/field_tests/movement_robot_field_test.py (hunk 2 above)
Comment (someone2060) [reply src/software/field_tests/movement_robot_field_test.py]: fixed

File: src/software/field_tests/passing_field_test.py [hunk 3]
```diff
@@ -96,11 +96,11 @@ def test_passing(field_test_runner):
         test_timeout_s=5,
     )
 
-    # Send a stop tactic after the test finishes
-    stop_tactic = StopTactic()
+    # Send a halt tactic after the test finishes
+    halt_tactic = StopTactic()
```
Comment (williamckha) [inline src/software/field_tests/passing_field_test.py]: This should be `HaltTactic`

File: src/software/ai/hl/stp/play/play.cpp [hunk 4]
```diff
@@ -192,7 +192,7 @@ std::unique_ptr<TbotsProto::PrimitiveSet> Play::get(
                 // StopTactics
```
Comment (williamckha) [inline src/software/ai/hl/stp/play/play.cpp]:
```suggestion
                // HaltTactics
```

File: src/software/ai/hl/stp/play/play.h [hunk 5]
```diff
@@ -171,7 +171,7 @@ class Play
                                 const WorldPtr& world_ptr) = 0;
 
     // Stop tactic common to all plays for robots that don't have tactics assigned
```
Comment (williamckha) [inline src/software/ai/hl/stp/play/play.h]:
```suggestion
    // HaltTactic common to all plays for robots that don't have tactics assigned
```

File: src/software/field_tests/passing_field_test.py [hunk 6]
```diff
@@ -96,11 +96,11 @@ def test_passing(field_test_runner):
...
     params = AssignedTacticPlayControlParams()
-    params.assigned_tactics[passer_robot_id].stop.CopyFrom(stop_tactic)
-    params.assigned_tactics[receiver_robot_id].stop.CopyFrom(stop_tactic)
+    params.assigned_tactics[passer_robot_id].stop.CopyFrom(halt_tactic)
+    params.assigned_tactics[receiver_robot_id].stop.CopyFrom(halt_tactic)
     # send the stop tactic
```
Comment (williamckha) [inline src/software/field_tests/passing_field_test.py]: We can just remove this comment, the code is self explanatory
```suggestion

```

**Comment by williamckha** [comment]
Looking good! just a couple nits

**Review by williamckha — DISMISSED** [review]
:100:

**Review by sauravbanna — DISMISSED** [review]
LGTM!

**Review by sauravbanna — APPROVED** [review]

**Review by williamckha — APPROVED** [review]

**Pull request merged**